test/test_sparse_csr.py

```diff
@@ -3764,12 +3764,12 @@ def broadcast_input(*ts):
                 if scale is None and query.size(-1) == 0:
                     scale = 1
                 expected = torch.nn.functional.scaled_dot_product_attention(
-                    *broadcast_input(query, key, value, attn_mask), scale=scale
+                    *broadcast_input(query.double(), key.double(), value.double(), attn_mask), scale=scale
                 )
```

Comment:
nit: let's probably add a comment here and do `expected = (..., q.double(), k.double(), value.double(),...).to(dtype)`?